### PR TITLE
Add the `whiteboard` option for `beaker` provision

### PR DESCRIPTION
**Pull request by idorax**
Pull Request Checklist

* [x] implement the feature
* [x] update the specification
* [x] modify the json schema
* [x] mention the version

Fixes: #2447

### CODE REVIEW

**Comment by guoguojenna** [comment]
It works for me, thanks.

https://beaker.engineering.redhat.com/jobs/8526651

$ tmt run prepare -h install -p wget discover -h fmf provision -h beaker -w "rdma pyverbs tests" --image RHEL-9.4.0-20231107.12 --hardware hostname=rdma-qe-39.rdma.lab.eng.rdu2.redhat.com  execute -h tmt tests -n "rdma/pyverbs-tests-2" plans --default -d -vvv
/var/tmp/tmt/run-005
Workdir '/var/tmp/tmt/run-005/plans/default' created.

/plans/default
    Workdir '/var/tmp/tmt/run-005/plans/default/discover' created.
    Workdir '/var/tmp/tmt/run-005/plans/default/provision' created.
    Workdir '/var/tmp/tmt/run-005/plans/default/prepare' created.
    Workdir '/var/tmp/tmt/run-005/plans/default/execute' created.
    Workdir '/var/tmp/tmt/run-005/plans/default/report' created.
    Report step always force mode enabled.
    Workdir '/var/tmp/tmt/run-005/plans/default/report' created.
    Workdir '/var/tmp/tmt/run-005/plans/default/finish' created.
    discover
    workdir: /var/tmp/tmt/run-005/plans/default/discover
        how: fmf
        order: 50
        Workdir '/var/tmp/tmt/run-005/plans/default/discover/default-0' created.
        directory: /home/zguo/Tests/kernel-tests
        Copy '/home/zguo/Tests/kernel-tests' to '/var/tmp/tmt/run-005/plans/default/discover/default-0/tests'.
        hash: 2f85eb36
        Check metadata tree in '/var/tmp/tmt/run-005/plans/default/discover/default-0/tests'.
        summary: 1 test selected
            /rdma/pyverbs-tests-2
    provision
    workdir: /var/tmp/tmt/run-005/plans/default/provision
        Workdir '/var/tmp/tmt/run-005/plans/default/provision/default-0' created.
        how: beaker
        order: 50
        hardware: hostname: = rdma-qe-39.rdma.lab.eng.rdu2.redhat.com

        whiteboard: rdma pyverbs tests
        image: RHEL-9.4.0-20231107.12
        Transformed hardware:
            and:
              - hostname:
                    _op: ==
                    _value: rdma-qe-39.rdma.lab.eng.rdu2.redhat.com
        guest: has been requested
        job id: 8526651

File: tmt/steps/provision/mrack.py [hunk 1]
```diff
@@ -508,6 +515,7 @@ class GuestBeaker(tmt.steps.provision.GuestSsh):
     _data_class = BeakerGuestData
 
     # Guest request properties
+    whiteboard: str
```
Comment (happz) [inline tmt/steps/provision/mrack.py:518]: `whiteboard` in `BeakerGuestData` is annotated as `Optional[str]`, this `whiteboard` needs to match the annotation.

Comment (idorax) [reply tmt/steps/provision/mrack.py]: Thanks for review. It's updated :-)

**Comment by guoguojenna** [comment]
https://beaker.engineering.redhat.com/jobs/8528170

$ tmt run discover -h fmf provision -h beaker --image RHEL-9.4.0-20231107.12 --hardware hostname=rdma-qe-39.rdma.lab.eng.rdu2.redhat.com  execute -h tmt tests -n "rdma/pyverbs-tests-2" plans --default -d -vvv
/var/tmp/tmt/run-008
Workdir '/var/tmp/tmt/run-008/plans/default' created.

/plans/default
    Workdir '/var/tmp/tmt/run-008/plans/default/discover' created.
    Workdir '/var/tmp/tmt/run-008/plans/default/provision' created.
    Workdir '/var/tmp/tmt/run-008/plans/default/prepare' created.
    Workdir '/var/tmp/tmt/run-008/plans/default/execute' created.
    Workdir '/var/tmp/tmt/run-008/plans/default/report' created.
    Report step always force mode enabled.
    Workdir '/var/tmp/tmt/run-008/plans/default/report' created.
    Workdir '/var/tmp/tmt/run-008/plans/default/finish' created.
    discover
    workdir: /var/tmp/tmt/run-008/plans/default/discover
        how: fmf
        order: 50
        Workdir '/var/tmp/tmt/run-008/plans/default/discover/default-0' created.
        directory: /home/zguo/Tests/kernel-tests
        Copy '/home/zguo/Tests/kernel-tests' to '/var/tmp/tmt/run-008/plans/default/discover/default-0/tests'.
        hash: 2f85eb36
        Check metadata tree in '/var/tmp/tmt/run-008/plans/default/discover/default-0/tests'.
        summary: 1 test selected
            /rdma/pyverbs-tests-2
    provision
    workdir: /var/tmp/tmt/run-008/plans/default/provision
        Workdir '/var/tmp/tmt/run-008/plans/default/provision/default-0' created.
        how: beaker
        order: 50
        hardware: hostname: = rdma-qe-39.rdma.lab.eng.rdu2.redhat.com

        image: RHEL-9.4.0-20231107.12
        Transformed hardware:
            and:
              - hostname:
                    _op: ==
                    _value: rdma-qe-39.rdma.lab.eng.rdu2.redhat.com
        guest: has been requested
        job id: 8528170

**Comment by idorax** [comment]
Thank you very much for helping to test this patch, @guoguojenna!

Right here let me have a simple summary based on the test results,
* If `-w <whiteboad>` is used, the beaker job will have the whiteboard set by user, e.g.  [8526651](https://beaker.engineering.redhat.com/jobs/8526651)
  ```bash
  tmt run prepare  -h install -p wget \
          discover -h fmf provision -h beaker   \
               -w "rdma pyverbs tests"  \
               --image RHEL-9.4.0-20231107.12 \
               --hardware hostname=rdma-qe-39.rdma.lab.eng.rdu2.redhat.com  \
          execute -h tmt tests -n "rdma/pyverbs-tests-2" plans --default -d -vvv
  ```
* If `-w <whiteboad>` is not used, the beaker job will have the whiteboard set by tmt (i.e. `tmt_name`), e.g. [8528170](https://beaker.engineering.redhat.com/jobs/8528170)
  ```bash
  tmt run discover -h fmf provision -h beaker \
               --image RHEL-9.4.0-20231107.12 \
               --hardware hostname=rdma-qe-39.rdma.lab.eng.rdu2.redhat.com  \
          execute -h tmt tests -n "rdma/pyverbs-tests-2" plans --default -d -vvv
  ```

**Review by lukaszachy — APPROVED** [review]

**Review by happz — APPROVED** [review]

File: tmt/steps/provision/mrack.py [hunk 2]
```diff
@@ -358,6 +359,12 @@ class BeakerGuestData(tmt.steps.provision.GuestSshData):
         )
 
     # Guest request properties
+    whiteboard: Optional[str] = field(
+        default=None,
+        option=('-w', '--whiteboard'),
+        metavar='WHITEBOARD',
+        help='Whiteboard of beaker job.'
```
Comment (psss) [inline tmt/steps/provision/mrack.py:366]: The help message does not say much, what about something like `Text describing the job, displayed in the list of jobs` or similar?

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for the improvement! Looks good, could we just enhance the help message a bit?

**Review by psss — APPROVED** [review]

**Comment by psss** [comment]
Added a short mention about `whiteboard` to the spec: fb36732e

**Comment by psss** [comment]
And schema adjustment in d5e18b1f.

**Pull request merged**